libavif.pc.cmake

```diff
@@ -0,0 +1,5 @@
+Name: ${PROJECT_NAME}
+Description: Library for encoding and decoding .avif files
```

Comment:
Perhaps you can try to add a `Requires:` and `Libs.private:` and other sections for when linking and compiling libavif etc so we can tell which encoder and decoder libavif was linked with

Reply:
I think this is a task for a package manager to make sure dependencies are correctly installed :-)

Reply:
I think this would be a task for the cmake list to put the line for
```bash
Requires: aom
```
if compiled with `AVIF_CODEC_AOM` instead of the packager to append that line every time to be able to use the pkg-config file for other projects since without it, if the user wants to compile something with libaif, they have to guess which one libaif was compiled with and pass `$(pkg-config --libs libavif [aom rav1e or dav1d])` to the libraries every time they want to compile instead of letting autoconf or others to find the libraries

Reply:
I'm assuming it's something like
```cmake
if(AVIF_CODEC_AOM)
    set(pkg_config_requires "aom ${pkg_config_requires}")
endif()
if(AVIF_CODEC_DAV1D)
    set(pkg_config_requires "dav1d ${pkg_config_requires}")
endif()
if(AVIF_CODEC_RAV1E)
    set(pkg_config_requires "rav1e ${pkg_config_requires}")
endif()
# Idk what it is for libgav1
```
and adding to the pc file
```pkg-config
Requires: @pkg_config_requires@
```

Reply:
I'm not entirely sure if it belongs in `Requires:` or `Requires.private:` since I mainly build using static libs and almost never shared and am mainly in the position of needing to sed almost all of the pc files since they never declare their dependencies properly.

Reply:
What if aom is statically linked into libavif? Then the Require would show up ...

Reply:
Also I don't see why we need a require for building against libavif. We do not need any header from aom, rav1e or dav1d.

Reply:
If libavif is statically linked to libaom, will libavif not require libaom when linking an executable?

Reply:
Compiling libavif.a myself and running nm shows that the aom, dav1d, and rav1e symbols as undefined

Reply:
attempting to compiling `avif_examples.c` manually `$CC ../examples/avif_example1.c -I../include -L. -lavif` (similar to what would be produced by `$(pkg-config --libs libavif)` when using the pkg-config file here)
produces
```none
ld.exe: ./libavif.a(codec_dav1d.c.o):codec_dav1d.c:(.text+0x22): undefined reference to `dav1d_close'
...
ld.exe: ./libavif.a(codec_rav1e.c.o):codec_rav1e.c:(.text+0x99): undefined reference to `rav1e_config_default'
...
ld.exe: ./libavif.a(codec_aom.c.o):codec_aom.c:(.text+0x25): undefined reference to `aom_codec_destroy'
...
```
(stripped down to reduce clutter)